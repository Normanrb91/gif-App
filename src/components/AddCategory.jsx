import PropTypes from 'prop-types'
import { useForm } from '../hooks/useForm'

export const AddCategory = ({ onNewCategory }) => {
  const [values, handleInputChange, reset] = useForm({ category: '' })

  const handleSubmit = e => {
    e.preventDefault()
    if (values.category.trim().length <= 1) return

    onNewCategory(values.category)
    reset()
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        placeholder="Buscar Gifs"
        name="category"
        type="text"
        value={values.category}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
