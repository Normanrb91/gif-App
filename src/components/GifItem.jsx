export const GifItem = ({ gif }) => {
  return <div>{gif.title}</div>
}
