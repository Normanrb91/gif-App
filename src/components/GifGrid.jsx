import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>

      {loading ? (
        <h2>Cargando...</h2>
      ) : (
        <div className="card-grid">
          {data.map(gif => (
            <GifItem key={gif.id} {...gif} />
          ))}
        </div>
      )}
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
