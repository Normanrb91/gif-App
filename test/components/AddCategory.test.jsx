import { fireEvent, render, screen } from '@testing-library/react'
import { AddCategory } from '../../src/components/AddCategory'

describe('pruebas en <AddCategory /> ', () => {
  test('debe hacer match con snapshot', () => {
    const { container } = render(<AddCategory onNewCategory={() => {}} />)
    expect(container).toMatchSnapshot()
  })

  test('debe de cambiar el valor del input', () => {
    render(<AddCategory onNewCategory={() => {}} />)

    const input = screen.getByRole('textbox')

    fireEvent.input(input, { target: { value: 'Betis' } })
    expect(input.value).toBe('Betis')
  })

  test('debe de llamar onNewCategory si el input tiene valor', () => {
    const inputValue = 'Betis'
    const onNewCategory = jest.fn()

    render(<AddCategory onNewCategory={onNewCategory} />)

    const input = screen.getByRole('textbox')
    const form = screen.getByRole('form')

    fireEvent.input(input, { target: { value: inputValue } })
    fireEvent.submit(form)

    expect(onNewCategory).toHaveBeenCalled()
    expect(onNewCategory).toHaveBeenCalledTimes(1)
    expect(onNewCategory).toHaveBeenCalledWith(inputValue)
    expect(input.value).toBe('')
  })

  test('no debe de llamar onNewCategory si el input esta vacio', () => {
    const onNewCategory = jest.fn()
    render(<AddCategory onNewCategory={onNewCategory} />)

    const form = screen.getByRole('form')
    fireEvent.submit(form)

    expect(onNewCategory).not.toHaveBeenCalled()
  })
})
