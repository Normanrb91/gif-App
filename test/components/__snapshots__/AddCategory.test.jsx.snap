// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AddCategory />  debe hacer match con snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      name="category"
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
